### DISCUSSION TITLE
What are the exact reasons a node can fail?

### DISCUSSION THREAD

**Discussion by StreetCube** [zwave-js/zwave-js#4774] [General] [answered]
I think the question is pretty self-explanatory. It would be useful for handling failed nodes to know the exact reason they can fail and I haven't found this information anywhere.
 Also I looked into "failing" nodes purposely for testing purposes but haven't found a way yet to do that either.

**Comment by AlCalzone** [comment, marked as answer]
Depends what you consider a failed node.
The Z-Wave controller considers a node as failed if it failed to acknowledge a command - this can happen sporadically when the connection is flaky or when the node actually has no power. Or when a battery-powered node sleeps.

Z-Wave JS estimates the node's status and can determine a node to be "dead". This is more strict than the "failed" status, because it will retry failed communication before changing the status, and it won't ever consider a sleeping node dead.
